core/src/main/java/io/opencensus/tags/TagContext.java

```diff
@@ -47,30 +49,39 @@
   }
 
   /**
-   * Returns a builder based on this {@code TagContext}.
+   * Returns the current {@code TagContext}.
    *
-   * @return a builder based on this {@code TagContext}.
+   * @return the current {@code TagContext}.
    */
-  public Builder toBuilder() {
-    return new Builder(getTags());
+  public static TagContext getCurrentTags() {
+    return CurrentTagsUtils.getCurrentTagContext();
   }
 
   /**
-   * Returns an empty {@code TagContext}.
+   * Returns a new empty {@code Builder}.
    *
-   * @return an empty {@code TagContext}.
+   * @return a new empty {@code Builder}.
    */
-  public static TagContext empty() {
-    return EMPTY;
+  public static Builder emptyBuilder() {
```

Comment:
So I agree and I did not say to remove it, it is just a proposal, we have the same issues in tracing and we want to a different approach where we do not have an explicit startSpanRoot. In my opinion an API like this:

```
public abstract class Tagger {
  // Returns the current TagContext from the Context or TagContext.Empty if nothing available in the Context.
  public TagContext getCurrentTagContext();

  // Starts a builder from the current tag context.
  public TagContextBuilder buildTagContext() {
    return buildTagContext(getCurrentTagContext());
  }

  // Starts a builder from the given tag context. If null then empty builder.
  public abstract TagContextBuilder buildTagContext(TagContext parent);
}
```

This way we make the API look very close to trace API + default implementation of the Tagger should return a NoopTagContextBuilder which does nothing (0 cost for not having the library installed).

Then you will need a class with a global state that allows access to the implementation: 
```
public final class Tagging {
  // loads with reflection the TagsComponent or TaggerImpl if we are sure we do not need to add more things. 
  private final Tagger tagger = Provider(...);

  public Tagger getTagger() {
    return tagger;
  }
}
```

Users will use this similar with Logger and Tracer in their class:

```
class MyClass {
  private static final Tagger tagger = Tagging.getTagger();

   ....
}
```